testing/BUILD

```diff
@@ -0,0 +1,21 @@
+load("//testing:bmv2_parsing_check.bzl", "subset_diff_test")
+
+test_suite(
+    name = "parsing",
+    tests = [
+      ":table_test",
+      ":basic_test"
+    ]
+)
+
+subset_diff_test(
+    name = "table_test",
```

Comment:
Another problem is that this is not going from a P4 program all the way to a protobuf, but I think that is what we would want, is it not?

Here's what I'm envisioning:

```
diff_test(
    name = "reflector_test",
    actual = ":reflector",
    expected = "testdata/reflector.expected",
)

run_parser(
    name = "reflector",
    src = "testdata/reflector.p4",
)
```

There is a rule `run_parser` that takes a P4 program as input and runs the parser, and outputs the protobuf. Then, based on that, we can have a diff-test that compares against the expected file.

We can still do subset diffing if we want as part of the run_parser rule, which would output some error in that case instead of (or maybe in addition to) the protobuf. But if subset diffing works, then the output should be the protobuf.

What do you think?

Reply:
Yes, sounds like a good idea, I will find a way to implement it